src/main/java/com/aws/greengrass/mqttclient/spool/Spool.java

```diff
@@ -81,11 +90,43 @@ private void setSpoolerConfigFromDeviceConfig(Topics topics) {
     private CloudMessageSpool setupSpooler() {
         if (config.getStorageType() == SpoolerStorageType.Memory) {
             return new InMemorySpool();
+        } else if (config.getStorageType() == SpoolerStorageType.FileSystem) {
+            try {
+                return getPersistenceSpoolGGService();
+            } catch (ServiceLoadException e) {
+                //log and use InMemorySpool
+                logger.atWarn()
+                        .kv(GG_PERSISTENCE_SPOOL_SERVICE_NAME_KEY, config.getPersistenceSpoolServiceName())
+                        .kv("ServiceLoadException", e.getMessage())
+                        .log("Persistence Spool set up failure, "
+                                + "defaulting to In-Memory Spool, retry when spooler accessed again.");
+                return new InMemorySpool();
+            } catch (InterruptedException e) {
+                throw new RuntimeException(e);
```

Comment:
this is not the way to handle an interruption.

Do:

`Thread.currentThread().interrupt();`